### DISCUSSION TITLE
Out of memory exception

### DISCUSSION THREAD

**Discussion by Devansh111** [LemonAppDev/konsist#896] [Ask a Question]
### Wanted end result.

I have been trying to implement konsist in my project, and for a simple example [test like this ](https://github.com/LemonAppDev/konsist?tab=readme-ov-file#android-specific-check), it is throwing an `java.lang.OutOfMemoryError: Java heap space` error, even with the heap size increased to 12g.. Debugging shows it happens only at `.classes()`. Is there a cap or a limit for how many files/classes konsist would work fine with?

### What you tried so far.

Tried using `.files` instead and somehow that doesn't cause OOM error but doesn't also give the required test check

**Comment by igorwojda** [comment]
Interesting. There is no cap limit. How many classes in protect do you have?

**Reply by Devansh111** [reply]
About 8k

**Reply by igorwojda** [reply]
I was able to reproduce this. I am looking into this.
https://lemonappdev.atlassian.net/browse/KON-613

**Reply by Devansh111** [reply]
Thank you

**Comment by igorwojda** [comment]
@Devansh111 I did some digging. Try increasing the `maxHeapSize` for the `test` source set and let me know if this helped:
https://docs.konsist.lemonappdev.com/help/troubleshooting/java.lang.outofmemoryerror-java-heap-space

**Reply by Devansh111** [reply]
@igorwojda Hey that worked! I added this instead and it worked - 
`tasks.withType(Test).configureEach {
    maxHeapSize = "1g"
}`

**Comment by igorwojda** [comment]
Thx for confirming. I have updated the Gradle Groovy docs.
